R/latex.R

```diff
@@ -230,9 +230,15 @@ add_toc_bib = function(x) {
 restore_block2 = function(x, global = FALSE) {
   i = grep('^\\\\begin\\{document\\}', x)[1]
   if (is.na(i)) return(x)
-  # add the necessary definition in the preamble when block2 engine (\BeginKnitrBlock) or pandoc
-  # fenced div (\begin) is used
-  if (length(grep(sprintf('^\\\\(BeginKnitrBlock|begin)\\{(%s)\\}', paste(all_math_env, collapse = '|')), x)) &&
+  # add the necessary definition in the preamble when block2 engine
+  # (\BeginKnitrBlock) or pandoc fenced div (\begin) is used if not already
+  # define. But don't do it with beamer and it defines already amsthm
+  # environments.
+  # An options allow external format to skip this part
+  # (useful for rticles see rstudio/bookdown#1001)
+  if (!xfun::isFALSE(getOption("bookdown.theorem.preamble", TRUE)) &&
```

Comment:
Usually I don't use `isTRUE()` or `isFALSE()` on `getOption()`. If users set a strange option value that is not `TRUE` or `FALSE`, I'll just let R signal the error naturally so users know that they made a mistake.

```suggestion
  if (getOption("bookdown.theorem.preamble", TRUE) &&
```

BTW, did you mean `!` before `getOption()`? It feels like we want to skip adding the preamble when `options(bookdown.theorem.preamble = FALSE)`.

Reply:
> If users set a strange option value that is not TRUE or FALSE, I'll just let R signal the error naturally so users know that they made a mistake.

Oh ok. I'll try to remember.

And yes we want to skip adding it when it is FALSE. thanks for catching that!